services/webnn/ort/graph_builder_ort.cc

```diff
@@ -2220,79 +2261,183 @@ GraphBuilderOrt::AddLayerNormalizationOperation(
     return base::ok();
   }
 
-  // TODO: crbug.com/356905058: Figure out if unordered axes should be allowed.
-  if (!std::ranges::is_sorted(axes)) {
-    return NewNotSupportedError("Axes must be ordered for layerNormalization.");
-  }
   const auto axes_size = axes.size();
+  // Sort the indexes of the elements in the axes array based on their values
+  // and return the sorted index array for adding a transpose operation if
+  // needed. For example input shape is [2, 1, 4, 3], the shape of the scale and
+  // bias is [3, 1, 4] if axes is [3, 1, 2], the sorted axes would be [1, 2, 3],
+  // then the permutation would be (sorted indices array) [1, 2, 0].
+  std::vector<uint32_t> permutation(axes_size);
```

Comment:
Make permutation optional? If the axes is consecutive, the transpose is unnecessary.

Reply:
> If the axes is consecutive, the transpose is unnecessary.

Agree. Added `if (!std::ranges::is_sorted(permutation))` before append transpose.